scripts/Overton/Overton_spacy.py

```diff
@@ -0,0 +1,118 @@
+import pandas as pd
+import spacy
+from tqdm import tqdm 
+import typer
+
+tqdm.pandas()
+
+
+def main(in_path: str, out_path:str, batch_size:int =10, spacy_model_name: str='en_core_web_md', sentences_embedding:str = 'all-MiniLM-L6-v2'):
+    """
+    This script takes output of overton preprocessing and prasing and creates basic spacy objects: nouns, noun chunks and lemmas for documents and paragraphs
+    """
+
+    def process(text):
+        try:
+            return en(text)
+        except Exception:
+            print("Error: {}".format(text))
+            return ""
+            
+    def get_nouns(docs):
+        """
+        Desc:   Get nouns from docs list
+        Input:  docs list of Doc objects        
+        Output: List of nouns that are not stop words from all docs list.
+        """
+        nouns = [token.text
+              for token in docs
+              if (not token.is_stop and
+                  not token.is_punct and
+                  token.pos_ == "NOUN")]
+        return nouns
+
+    def get_chunks(docs, stopwords = False):
+        """
+        Desc:   Prepares list of noun chunks present in all docs
+        Input:  docs list of Doc objects 
+                stopwords bool describving if we want to remove stopwords or not
+        Output: list of noun chunks
+        """
+        noun_chunks = []
+        for chunk in docs.noun_chunks:
+            if stopwords:
+              stop = True
+              for w in chunk:
+                if w.is_stop:

```

Comment:
tu chcieliśmy wywalić noun chunki które mają w sobie stopwordy czy wywalić stopwordy z noun chunków?

Reply:
Jeżeli chunk ma stop worda w środku, to nie zapisujemy chunka. Ale to jest nieważne, bo i tak stopwords są na false w skrypcie